pkg/utils/utils.go

```diff
@@ -115,20 +115,19 @@ func CustomizeServiceAccount(logger Logger, sa *corev1.ServiceAccount, uiService
 	}
 	// Adding the OAuth proxy to the service account.
 	sa.Annotations[OAuthRedirectAnnotationName] = "{\"kind\":\"OAuthRedirectReference\",\"apiVersion\":\"v1\",\"reference\":{\"kind\":\"Route\",\"name\":\"" + uiService.GetName() + "\"}}"
-	// Adding pull secrets to the service account.
-	imagePullSecrets := make([]corev1.LocalObjectReference, 1)
-	imagePullSecrets[0] = corev1.LocalObjectReference{
-		Name: "sa-" + sa.GetNamespace(),
-	}
+	
+	// Retrieving pull secrets from kappnav CR and adding them to the service account.
 	pullSecrets := instance.Spec.Image.PullSecrets
```

Comment:
How does this work when the service account is initially created? What happens if sa.ImagePullSecrets is nil?

Reply:
I have confirmed with Artur and he said that once its submitted into k8s it will trigger watches, so whoever responds first will read original object and will react to it.  The append function will create new slice if original object is nil.  
The issue is caused by kappnav-sa always modifies imagePullSecrets by setting it to sa-kappnav, but open shift will inject its own secret (required for internal registry).  The data should be merged with a cluster not overwritten and it will create an infinite reconcile loop

Reply:
The array is accessed by containSecret() before append() is called. Did you test this with a kAppNav CR that contained secrets?

Reply:
yes

Reply:
image:
    pullPolicy: Always
    pullSecrets: ['kappnav-sa-dockercfg-zkknn', 'kappnav-sa-dockercfg-abcde']

Reply:
it merges successfully

Reply:
FYI, I checked with Artur and apparently it's okay to access a nil array with "range", but in general it's not a safe thing to be doing.